doc/dfu/quark_dfu_uart.rst

```diff
@@ -0,0 +1,140 @@
+Quark DFU over UART Guide
+#########################
+
+.. contents::
+
+Hardware Setup
+**************
+
+The hardware setup consists of two main steps:
+
+ * Connecting the board to a Linux host using UART.
+ * Having the possibility to connect the DM GPIO pin to GND or to press the DM
+   Button.
+
+The DM GPIO pin is the pin used to put the device into DM mode. The device has
+to be reset while the GPIO is connected to GND. On Quark Developer Boards a
+button connects the DM GPIO to GND if pressed.
+
```

Comment:
Do we need to mention WHICH GPIO's / Button.

SW2 on Intel® Quark™Developer Kit D2000 or SW1 on Intel® Quark™ SE Microcontroller C1000

AP_GPIO_AON4 for Intel® Quark™ SE Microcontroller C1000 Developer Board(pin J14.43)
F2_DIO_05 for Intel® Quark™ Developer Kit D2000 (pin J4.6)